iml-gui/crate/src/components/duration_picker.rs

```diff
@@ -0,0 +1,248 @@
+// Copyright (c) 2019 DDN. All rights reserved.
+// Use of this source code is governed by a MIT-style
+// license that can be found in th e LICENSE file.
+
+use super::super::environment::MAX_SAFE_INTEGER;
+use crate::{
+    components::{dropdown, font_awesome, tooltip, Placement},
+    extensions::MergeAttrs,
+    extensions::NodeExt as _,
+    generated::css_classes::C,
+};
+use seed::{prelude::*, *};
+use std::fmt;
+
+#[derive(Debug, Clone, Copy, PartialEq)]
+pub enum Unit {
+    Days,
+    Hours,
+    Minutes,
+    Seconds,
+}
+
+impl fmt::Display for Unit {
+    fn fmt(&self, f: &mut fmt::Formatter) -> fmt::Result {
+        write!(f, "{:?}", self)
```

Comment:
Display depends on Debug.

Reply:
I think Display can be derived.

Reply:
When I tried this I get the following error:
```
error: cannot find derive macro `Display` in this scope
  --> src/components/duration_picker.rs:14:17
   |
14 | #[derive(Debug, Display, Clone, Copy, PartialEq)]
```
There is a crate called `enum_display_derive` that would solve this problem but I don't think it's worth pulling in an extra dependency for this.